### ISSUE TITLE
Does not work with Rancher Desktop

### ISSUE THREAD

**Issue by 5H3PH3RDCSG** [microsoft/vscode-remote-release#6252]
Issue Type: <b>Feature Request</b>

Allow the use of Rancher Desktop instead of Docker Desktop.

Extension version: 0.209.6
VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:40:02.816Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

**Comment by gpyga** [comment]
@5H3PH3RDCSG, [this guide](https://docs.rancherdesktop.io/tutorials/vs-code-remote-containers) is new as of raising this issue. Looks like it was committed Feb 14th, 2022. The gist is that you have to be using the `dockerd` runtime.

**Comment by Chuxel** [comment]
The link changed, but @gpyga 's point is right.  https://docs.rancherdesktop.io/how-to-guides/vs-code-remote-containers

Also, nerdctl support is tracked as #6014. https://github.com/microsoft/dev-container-spec/issues/11 and https://github.com/microsoft/dev-container-spec/issues/9 are also relevant for community contributions here. With that in mind, I'd propose we close this as resolved.

**Comment by muziekklas** [comment]
I've just tried using nerdctl instead the dockercli in remote containers and it fails during checking the version of the docker api:
level=fatal msg="template: :1:9: executing \"\" at <.Server.APIVersion>: can't evaluate field APIVersion in type *dockercompat.ServerVersion"
After some trail & error, I could fetch the server version with the following command:
nerdctl version --format '{{(index .Server.Components 0).Version}}'

After this, I've done some digging in moby's source code compared to the nerdctl sourcecode and I've noticed that the Properties .Server.Version are marked deprecated and are kept for backwards compat.:
https://github.com/moby/moby/blob/2ed904cad7055847796433cc56ef1d1de0da868c/api/types/types.go#L219
The nerdctl guys did not implement it in their implementation: https://github.com/containerd/nerdctl/blob/2b4e3bde28d67d08c85bcf15fef23d973bae590f/pkg/inspecttypes/dockercompat/info.go#L95.

I wonder, whether you guys could add extra configuration entries like Version format template (so we can configure the version template thingy) and also making the minimum version configurable, I could try using Devcontainers + nerdctl.

Or, alternatively give users the options to disable versionchecks all together and just hint out to users that the devcontainers might not work 100% or something. 
Just throwing out some ideas...
